pkg/cli/mirror/catalog_images.go

```diff
@@ -160,87 +140,30 @@ func (o *MirrorOptions) processCatalogRefs(ctx context.Context, catalogsByImage
 		// Check push permissions before trying to resolve for Quay compatibility
 		nameOpts := getNameOpts(destInsecure)
 		remoteOpts := getRemoteOpts(ctx, destInsecure)
-		ref, err := name.ParseReference(refExact, nameOpts...)
-		if err != nil {
-			return err
-		}
-		err = remote.CheckPushPermission(ref, authn.DefaultKeychain, createRT(destInsecure))
-		if err != nil {
-			return err
-		}
 		imgBuilder := &builder.ImageBuilder{
 			NameOpts:   nameOpts,
 			RemoteOpts: remoteOpts,
 		}
 
-		switch _, _, rerr := resolver.Resolve(ctx, refExact); {
-		case errors.Is(rerr, errdefs.ErrNotFound):
-			logrus.Infof("Catalog image %q found, building from file with file-based catalog ", refExact)
-
-			add, err := builder.LayerFromPath("/configs", filepath.Join(artifactDir, IndexDir, "index.json"))
-			if err != nil {
-				return fmt.Errorf("error creating add layer: %v", err)
-			}
-
-			// Since we are defining the FBC as index.json, remove
-			// any .yaml files from the initial image to ensure they are not processed instead
-			deleted, err := deleteLayer("/configs/.wh.index.yaml")
-			if err != nil {
-				return fmt.Errorf("error creating deleted layer: %v", err)
-			}
-			layers = append(layers, add, deleted)
-
-			layoutDir := filepath.Join(artifactDir, LayoutsDir)
-			layoutPath, err = imgBuilder.CreateLayout("", layoutDir)
-			if err != nil {
-				return fmt.Errorf("error creating OCI layout: %v", err)
-			}
-		case rerr == nil:
-			logrus.Infof("Catalog image %q found, rendering with file-based catalog", refExact)
+		logrus.Infof("Rendering catalog image %q with file-based catalog ", refExact)
 
-			dcDir := filepath.Join(artifactDir, IndexDir)
-			dc, err := action.Render{
-				// Order the old ctlgRef before dcDir so new packages/channels/bundles overwrite
-				// existing counterparts.
-				Refs:           []string{refExact, dcDir},
-				AllowedRefMask: action.RefAll,
-				Registry:       reg,
-			}.Run(ctx)
-			if err != nil {
-				return err
-			}
-			// Remove any duplicate objects
-			merger := &operator.TwoWayStrategy{}
-			if err := merger.Merge(dc); err != nil {
-				return err
-			}
-			dcDirToBuild := filepath.Join(dcDir, "rendered")
-			if err := os.MkdirAll(dcDirToBuild, os.ModePerm); err != nil {
-				return err
-			}
-			renderedPath := filepath.Join(dcDirToBuild, "index.json")
-			f, err := os.Create(renderedPath)
-			if err != nil {
-				return err
-			}
-			defer f.Close()
-			if err := declcfg.WriteJSON(*dc, f); err != nil {
-				return err
-			}
+		add, err := builder.LayerFromPath("/configs", filepath.Join(artifactDir, config.IndexDir, "index.json"))
```

Comment:
It's not really something oc-mirror has control over but it always makes me slightly concerned that we have to rely on a filename to detect the FBC file and have to do the below cleanup to ensure we don't process wrong files. We need to ensure this `index.json` is something that the pipeline team will reliably use to produce their catalogsource images. Plus, technically OLM doesn't restrict to only one file per catalogsource. The catalogsource can have multiple FBC files which contains pieces of the whole FBC.

Reply:
I'm fine with what you are doing here. I just wanted to post my comment here as something to keep an eye on.

Reply:
@dinhxuanvu `oc-mirror` is actually generating the `index.json` so we shouldn't have a problem there. Totally valid concern though when you consider that multiple files under `/configs` could be aggregated with what we are producing. We do remove any `index.yaml` that may have been under `/configs` originally, but it may be safer to ensure `/configs` is empty and the only FBC in the image is the one being produced by `oc-mirror`.